fe/src/main/java/org/apache/doris/load/DeleteHandler.java

```diff
@@ -509,8 +510,8 @@ private void checkDeleteV2(OlapTable table, Partition partition, List<Predicate>
                 if (column == null) {
                     ErrorReport.reportDdlException(ErrorCode.ERR_BAD_FIELD_ERROR, columnName, indexName);
                 }
-
-                if (table.getKeysType() == KeysType.DUP_KEYS && !column.isKey()) {
+                MaterializedIndexMeta indexMeta = table.getIndexIdToMeta().get(index.getId());
+                if (indexMeta.getKeysType() != KeysType.DUP_KEYS && !column.isKey()) {
```

Comment:
if index is DUPLICATED, but column type is float or double, we should also forbid the delete operation.